### ISSUE TITLE
Generating complete documentation: Doxygen and NUM_PROC_THREADS

### ISSUE THREAD

**Issue by phillipjohnston** [johnttaylor/pim#10]
The current Doxyfile is set to generate documentation for `src/Cpl` only. I wanted to get this running with at least `src/Storm` as well so I could get more useful inheritance and collaboration diagrams to see how the different pieces of the ecosystem fit together.

I did the simplest thing first and tried to run Doxygen on the entire source tree. This caused segmentation faults, or it caused weird "could not open file for reading/writing" errors. Then I just added the additional Storm directory along with Cpl. Same result.

Then I tried Storm alone, and that worked just fine -  most of the time!

After a bit of debugging, I realized the issue I was experiencing came down to the NUM_PROC_THREADS setting. Setting this back to "1" made things take longer, sure, but documentation for the whole source tree generates successfully without errors. Seems like Doxygen has some multithreading race condition here that is occasionally blowing up the process.

```
# Minimum value: 0, maximum value: 32, default value: 1.
NUM_PROC_THREADS       = 1
```

**Comment by johnttaylor** [comment]
I am not seeing the Doxygen errors.  I modified the `top\Doxyfile` to include the `src\Storm` directory - and doxygen ran fine - except it generated a few warnings.  I will update the Doxyfile file and fix the Doxygen warnings.  Below is the file snippet of where I added the Storm directory to the Doxyfile:

```
# The INPUT tag is used to specify the files and/or directories that contain
# documented source files. You may enter file names like myfile.cpp or
# directories like /usr/src/myproject. Separate the files or directories with
# spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
# Note: If this tag is empty the current directory is searched.

INPUT                  = ../src/Cpl  \
                         ../src/Storm
```

The attached file is screenshot of the Doxygen output:
![image](https://user-images.githubusercontent.com/10332392/178063589-df8ffda2-08e7-4a99-a9fd-4f5f31eb78d1.png)

**Comment by phillipjohnston** [comment]
It fails 100% of the time for me with the equivalent change in the Doxyfile. Only changing the thread count back to 1 gets it to work. [Others have run into this kind of thing before too](https://github.com/doxygen/doxygen/issues/8090#issuecomment-1076726865). 

Sample errors included below. There are more variations on this theme.

```
Generating code for file Cpl/Persistent/IndexedEntryReader.h...
Generating code for file Cpl/Persistent/IndexedEntryRecord.h...
error: Could not open file /Users/phillip/src/ea/pim/top/../docs/html/_pi_pre_process_8h_source.html for writing
zsh: segmentation fault  doxygen
```

```
Generating code for file Storm/Type/OperatingMode.h...
error: could not open file /Users/phillip/src/ea/pim/src/Storm/Type/OperatingMode.h
error: cannot open file '/Users/phillip/src/ea/pim/src/Storm/Type/OperatingMode.h' for reading
Generating code for file Storm/Component/Pi.h...
error: Could not open file /Users/phillip/src/ea/pim/top/../docs/html/_pi_pre_process_8h_source.html for writing
zsh: segmentation fault  doxygen
```

```
Generating code for file Storm/Component/HvacRelayOutputs.h...
error: could not open file /Users/phillip/src/ea/pim/src/Storm/Component/HvacRelayOutputs.h
error: cannot open file '/Users/phillip/src/ea/pim/src/Storm/Component/HvacRelayOutputs.h' for reading
error: Could not open file /Users/phillip/src/ea/pim/top/../docs/html/_idt_selection_8h_source.html for writing
Generating code for file Storm/Type/HvacRelayOutputs.h...
zsh: segmentation fault  doxygen
```

**Comment by johnttaylor** [comment]
What version of Doxygen are you running?  I am running 1.9.2 on Windoze 11.  What host are running on (Windoze, Linux, MacOS)?

**Comment by phillipjohnston** [comment]
I'm running 1.9.3 on MacOS

**Comment by phillipjohnston** [comment]
Looks like there will be a number of fixes for race conditions in 1.9.4: https://github.com/doxygen/doxygen/issues/9155
